### PR TITLE
Mark previously deprecated SSL settings as obsolete

### PR DESCRIPTION
**Pull request by donoghuc**
- SSL settings that were marked deprecated in version `11.14.0` are now marked obsolete, and will prevent the plugin from starting.
  - These settings are:
    - `cacert`, which should be replaced by `ssl_certificate_authorities`
    - `keystore`, which should be replaced by `ssl_keystore_path`
    - `keystore_password`, which should be replaced by `ssl_keystore_password`
    - `ssl`, which should be replaced by `ssl_enabled`
    - `ssl_certificate_verification`, which should be replaced by `ssl_verification_mode`
    - `truststore`, which should be replaced by `ssl_truststore_path`
    - `truststore_password`, which should be replaced by `ssl_truststore_password`

Thanks for contributing to Logstash! If you haven't already signed our CLA, here's a handy link: https://www.elastic.co/contributor-agreement/

Closes https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/1190

### CODE REVIEW

**Review by robbavey — APPROVED** [review]
Code LGTM, over to @karenzone for docs approval

**Comment by donoghuc** [comment]
Corresponding LS deprecation docs PR https://github.com/elastic/logstash/pull/16787

**Comment by donoghuc** [comment]
I came up with a workflow to run the acceptance tests in LS against these changes un this branch https://github.com/elastic/logstash/tree/test-plugin-from-gh-source-buildkite-run CI runs https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/1958 are red but it appears to be due to installing a gem from GH source instead of ssl deprecation issues. So i think this should not cause any issues once it promotes into LS.

**Review by karenzone — COMMENTED** [review]
DO NOT MERGE:  We'll need to review and merge this fix (https://github.com/elastic/logstash/pull/16798) first to avoid breaking the docs build.

File: docs/index.asciidoc [hunk 1]
```diff
@@ -1324,99 +1324,20 @@ https://www.elastic.co/blog/elasticsearch-versioning-support[versioning support
...
-
-* Value type is <<password,password>>
-* There is no default value for this setting.
-
-Set the truststore password
-
-[id="plugins-{type}s-{plugin}-common-options"]
-include::{include_path}/{type}.asciidoc[]
-
-:no_codec!:
```
Comment (karenzone) [inline docs/index.asciidoc:1422]: This needs to be at the end of every plugin doc file. 
Please add it back. :-)

Comment (donoghuc) [reply docs/index.asciidoc]: OH, wow, yeah my bad! Thanks for catching that.

**Review by karenzone — COMMENTED** [review]
Line 328-329:  Please change to 

This plugin supports the these configuration options plus the <<plugins-{type}s {plugin}-common-options>> described later.

To resolve: 
- bad link causing docs build to fail
- inaccurate information (that we're continue to support deleted settings)

**Review by karenzone — COMMENTED** [review]
Line 443:
Please change `<<plugins-{type}s-{plugin}-ssl,`ssl_enabled => true`>>` to 
`<<plugins-{type}s-{plugin}-ssl_enabled,`ssl_enabled => true`>>`. 

It's looking for the now deleted `ssl` setting. 

(GitHub won't allow me to comment on lines not included in this PR. )

**Comment by karenzone** [comment]
@donoghuc, after pulling down your latest update and making that last fix noted Line 443, I'm getting a passing doc build.  We're so close! 

Ugh... now we have some rendering problems.  I'll have those for you shortly.

File: docs/index.asciidoc [hunk 2]
```diff
@@ -1324,97 +1323,23 @@ https://www.elastic.co/blog/elasticsearch-versioning-support[versioning support
...
++[cols="<,<",options="header",]
+ |=======================================================================
++|Setting|Replaced by
++| cacert | <<plugins-{type}s-{plugin}-ssl_certificate_authorities>>
++| keystore | <<plugins-{type}s-{plugin}-ssl_keystore_path>>
++| keystore_password | <<plugins-{type}s-{plugin}-ssl_keystore_password>>
++| ssl | <<plugins-{type}s-{plugin}-ssl_enabled>>
++| ssl_certificate_verification | <<plugins-{type}s-{plugin}-ssl_verification_mode>>
++| truststore | <<plugins-{type}s-{plugin}-ssl_truststore_path>>
++| truststore_password | <<plugins-{type}s-{plugin}-ssl_truststore_password>>
```
Comment (karenzone) [inline docs/index.asciidoc:1341]:
```suggestion
WARNING: As of version `12.0.0` of this plugin, some configuration options have been replaced.
The plugin will fail to start if it contains any of these obsolete options.

+[cols="<,<",options="header",]
 |=======================================================================
|Setting|Replaced by
| cacert | <<plugins-{type}s-{plugin}-ssl_certificate_authorities>>
| keystore | <<plugins-{type}s-{plugin}-ssl_keystore_path>>
| keystore_password | <<plugins-{type}s-{plugin}-ssl_keystore_password>>
| ssl | <<plugins-{type}s-{plugin}-ssl_enabled>>
| ssl_certificate_verification | <<plugins-{type}s-{plugin}-ssl_verification_mode>>
| truststore | <<plugins-{type}s-{plugin}-ssl_truststore_path>>
| truststore_password | <<plugins-{type}s-{plugin}-ssl_truststore_password>>
```

File: docs/index.asciidoc [hunk 3]
```diff
@@ -1324,97 +1323,23 @@ https://www.elastic.co/blog/elasticsearch-versioning-support[versioning support
...
-[id="plugins-{type}s-{plugin}-truststore_password"]
-===== `truststore_password`
-deprecated[11.14.0, Replaced by <<plugins-{type}s-{plugin}-ssl_truststore_password>>]
-
-* Value type is <<password,password>>
-* There is no default value for this setting.
-
-Set the truststore password
+[id="plugins-{type}s-{plugin}-obsolete-options"]
++==== Elasticsearch Output Obsolete Configuration Options
```
Comment (karenzone) [inline docs/index.asciidoc:1327]:
```suggestion
==== Elasticsearch Output Obsolete Configuration Options
```

**Review by karenzone — COMMENTED** [review]
This change didn't get implemented and is causing the doc build to fail. 

https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1197#pullrequestreview-2516309146

**Comment by donoghuc** [comment]
> This change didn't get implemented and is causing the doc build to fail.
>
> https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1197#pullrequestreview-2516309146

Sorry, that got lost in the shuffle. Updated in https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1197/commits/7c4be65f69397a652c1e6933c0cbc2c5d7419d3e

**Comment by karenzone** [comment]
GitHub won't let me make suggestions to lines that didn't have changes.  That's the reason, I have to note these as general suggestions with Line numbers.

File: docs/index.asciidoc [hunk 4]
```diff
@@ -1324,97 +1323,23 @@ https://www.elastic.co/blog/elasticsearch-versioning-support[versioning support
...
-* There is no default value for this setting.
-
-Set the truststore password
+[id="plugins-{type}s-{plugin}-obsolete-options"]
+==== Elasticsearch Output Obsolete Configuration Options
+
+WARNING: As of version `12.0.0` of this plugin, some configuration options have been replaced.
+The plugin will fail to start if it contains any of these obsolete options.
+
+[cols="<,<",options="header",]
```
[the remark this answers is not in the document]
Comment (donoghuc) [reply docs/index.asciidoc:1332]: Updated in [8282103](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1197/commits/8282103e91d15cfeebadf47077d6a9ee9ff50df1)

File: docs/index.asciidoc [hunk 5]
```diff
@@ -1330,7 +1330,7 @@ WARNING: As of version `12.0.0` of this plugin, some configuration options have
 The plugin will fail to start if it contains any of these obsolete options.
 
 [cols="<,<",options="header",]
- |=======================================================================
+=======================================================================
```
Comment (karenzone) [inline docs/index.asciidoc:1333]: The pipe in Line 1333 needs to stay, please.

Comment (donoghuc) [reply docs/index.asciidoc]: 🤦 i'm really doing poorly today. Need more coffee.

File: docs/index.asciidoc [hunk 6]
```diff
@@ -325,8 +325,7 @@ When a string value on an event contains one or more byte sequences that are not
 [id="plugins-{type}s-{plugin}-options"]
 ==== Elasticsearch Output Configuration Options
 
-This plugin supports the following configuration options plus the
-<<plugins-{type}s-{plugin}-common-options>> and the <<plugins-{type}s-{plugin}-deprecated-options>> described later.
+This plugin supports these configuration options plus the <<plugins-{type}s-{plugin}-common-options>> described later.
```
Comment (karenzone) [inline docs/index.asciidoc:328]:
```suggestion
This plugin supports these configuration options plus the <<plugins-{type}s-{plugin}-common-options>> described later.

NOTE: As of version 12.0.0 of this plugin, a number of previously deprecated SSL settings have been removed.
Please check out <<plugins-{type}s-{plugin}-obsolete-options>> for details."
```

Comment (karenzone) [reply docs/index.asciidoc:328]: Content is building and rendering correctly.  We're missing this line that directs users to deleted content.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,15 @@
...
+    - `keystore`, which should be replaced by `ssl_keystore_path`
+    - `keystore_password`, which should be replaced by `ssl_keystore_password`
+    - `ssl`, which should be replaced by `ssl_enabled`
+    - `ssl_certificate_verification`, which should be replaced by `ssl_verification_mode`
+    - `truststore`, which should be replaced by `ssl_truststore_path`
+    - `truststore_password`, which should be replaced by `ssl_truststore_password`
+    - [#1197](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1197)
+
 ## 11.22.10
  - Add `x-elastic-product-origin` header to Elasticsearch requests [#1194](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1194)
```
Comment (karenzone) [inline CHANGELOG.md:15]: NO CHANGES NEEDED. Just a note...

ON MERGE, we might have a conflict to resolve.  Please be sure to keep what's currently in source.  
The correct PR number is #1195.

Comment (karenzone) [inline CHANGELOG.md:15]: NO CHANGE NEEDED HERE. 

ON MERGE, we might have a conflict to resolve. Please be sure to keep what's currently in source.
The correct PR number is https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1195.

Comment (donoghuc) [reply CHANGELOG.md:15]: I can explicitly rebase against main and push that to this branch. Locally that looks right, i'll push so there is no confusion.

Comment (donoghuc) [reply CHANGELOG.md:15]: Pushed the rebase, it looks correct in the expanded diff now :)

**Review by karenzone — APPROVED** [review]
Builds cleanly, renders as expected, and LGTM!  🚀

**Comment by donoghuc** [comment]
Amazing! thanks so much for all the help (and patience) on this one.

**Pull request merged**